tests/files/test-api-client.php

```diff
@@ -314,8 +322,19 @@ public function get_test_data__get_file() {
 	public function test__get_file( $mocked_response, $expected_result ) {
 		$this->mock_http_response( $mocked_response );
 
-		$actual_result = $this->api_client->get_file( '/wp-content/uploads/file.jpg' );
+		$file = $this->api_client->get_file( '/wp-content/uploads/get_file.jpg' );
+		
+		if ( is_object( $file ) ) {
```

Comment:
In what scenarios does `get_file()` return an object now - shouldn't the behavior be the same as before (and if not, are there any other spots that need to account for different types)?

Reply:
I think the Api_Client is also used in one of the WP_Filesystem_VIP classes so we should make sure to update those uses as well.

Reply:
Yeah, looks like here:

https://github.com/Automattic/vip-go-mu-plugins/blob/master/files/class-wp-filesystem-vip-uploads.php#L60-L68

We'll need to update that method to read the file resource and return the file contents (it's another point where OOM errors can happen but I we can't really change the method signature there).

Reply:
Right now it's returning a `WP_Error` object or the file stream. I'm not sure if this is the best way though. Perhaps throwing an error instead so that we have a consistent return type is a better idea.

Reply:
Ah a WP_Error is fine - in that case we should use `is_wp_error()` to detect it.

Reply:
@mjangda Updated in 354d9bb812247816b96539f64e785a5db61562b0